src/dumbo/src/ns.rs

```diff
@@ -221,10 +236,16 @@ impl MmdsNetworkStack {
     }
 
     fn write_arp_reply(
-        &self,
+        &mut self,
         buf: &mut [u8],
-        dst_ipv4: Ipv4Addr,
     ) -> Result<Option<NonZeroUsize>, WriteArpFrameError> {
+        let arp_reply_info_opt = self.pending_arp_reply_dest.first();
+        if arp_reply_info_opt.is_none() {
+            return Err(WriteArpFrameError::NoPendingArpReply);
+        }
+
+        // Safe to unwrap because it is checked before.
+        let arp_reply_info = arp_reply_info_opt.unwrap();
```

Comment:
This can be replaced with:
```rust
let arp_reply_info = self
            .pending_arp_reply_dest
            .first()
            .ok_or_else(|| WriteArpFrameError::NoPendingArpReply)?;
```